crates/nu-command/src/filesystem/touch.rs

```diff
@@ -75,88 +74,58 @@ impl Command for Touch {
             });
         }
 
-        let mut date: Option<DateTime<Local>> = None;
-        let mut ref_date_atime: Option<DateTime<Local>> = None;
+        let mut mtime = SystemTime::now();
+        let mut atime = SystemTime::now();
 
-        // Change both times if none is specified
+        // Change both times if neither is specified
         if !change_mtime && !change_atime {
             change_mtime = true;
             change_atime = true;
         }
 
-        if change_mtime || change_atime {
-            date = Some(Local::now());
-        }
-
-        if use_reference {
-            let reference: Option<Spanned<String>> =
-                call.get_flag(engine_state, stack, "reference")?;
-            match reference {
-                Some(reference) => {
-                    let reference_path = Path::new(&reference.item);
-                    if !reference_path.exists() {
-                        return Err(ShellError::TypeMismatch {
-                            err_message: "path provided is invalid".to_string(),
-                            span: reference.span,
-                        });
-                    }
-
-                    date = Some(
-                        reference_path
-                            .metadata()
-                            .expect("should be a valid path") // Should never fail as the path exists
-                            .modified()
-                            .expect("should have metadata") // This should always be valid as it is available on all nushell's supported platforms (Linux, Windows, MacOS)
-                            .into(),
-                    );
-
-                    ref_date_atime = Some(
-                        reference_path
-                            .metadata()
-                            .expect("should be a valid path") // Should never fail as the path exists
-                            .accessed()
-                            .expect("should have metadata") // This should always be valid as it is available on all nushell's supported platforms (Linux, Windows, MacOS)
-                            .into(),
-                    );
-                }
-                None => {
-                    return Err(ShellError::MissingParameter {
-                        param_name: "reference".to_string(),
-                        span: call.head,
-                    });
-                }
+        if let Some(reference) = reference {
+            let reference_path = Path::new(&reference.item);
+            if !reference_path.exists() {
+                return Err(ShellError::FileNotFoundCustom {
+                    msg: "Reference path not found".into(),
+                    span: reference.span,
+                });
             }
+
+            let metadata = reference_path.metadata().expect("should be a valid path"); // Should never fail as the path exists
```

Comment:
These `expect`s are the only thing that gives me pause about this PR. I've seen so many complaints about things like this panicking even though they shouldn't normally. I'd rather have some better error handling with these. Let's see what others think.

Reply:
I've tried `touch`ing a file when I don't have read perms. In that case the `.exists()` should return false so we wouldn't get to the `expect`.

Have there been cases where something seemingly infallible like that fails?

Reply:
mostly broken files or folders in messed up file systems. it's really hard to repro.